File: 1668-maximum-substring/1668.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	maxRepeating("ababc", "ab")
}

func maxRepeating(sequence string, word string) int {
	res := 0
	wordToTest := word

	for {
		if strings.Contains(sequence, wordToTest) {
			res++
			wordToTest += word
			continue
		}
		break
	}

	fmt.Println(res)
	return res
}

File: 1668-maximum-substring/1668.ts
function maxRepeating(sequence: string, word: string): number {
  let res = 0
  let test = word
  while (true) {
    if (sequence.includes(test)) {
      test += word
      res++
      continue
    }
    break
  }
  return res
}

maxRepeating('ababc', 'ab')
